auth0/data_source_auth0_client.go

```diff
@@ -10,17 +10,19 @@ import (
 )
 
 func newDataClient() *schema.Resource {
-	clientSchema := datasourceSchemaFromResourceSchema(newClient().Schema)
-	delete(clientSchema, "client_secret_rotation_trigger")
-
-	addOptionalFieldsToSchema(clientSchema, "name", "client_id")
-
 	return &schema.Resource{
 		Read:   readDataClient,
-		Schema: clientSchema,
+		Schema: newClientSchema(),
 	}
 }
 
+func newClientSchema() map[string]*schema.Schema {
+	clientSchema := datasourceSchemaFromResourceSchema(newClient().Schema)
+	delete(clientSchema, "client_secret_rotation_trigger")
+	addOptionalFieldsToSchema(clientSchema, "name", "client_id")
+	return clientSchema
+}
+
```

Comment:
Because both the client and global client data sources share the same schema, I needed to separate the schema creation into it's own reusable function.